### DISCUSSION TITLE
Visualization of PointStamped message

### DISCUSSION THREAD

**Discussion by vinnnyr** [foxglove/community#579] [Q&A] [answered]
Hello! I would like to put in a feature request to visualize PointStamped message in foxglove studio. 

This is helpful when relevant since it helps in debugging the Nav2 Regulated Pure Pursuit Controller, so I believe more people than just me would benefit from this enhancement.

**Comment by jtbandes** [comment, marked as answer]
In order to view these today, there are two options which will let you view PointStamped messages by converting them to another type (such as [SceneUpdate](https://foxglove.dev/docs/studio/messages/scene-update)):
- Create a [User Script](https://foxglove.dev/docs/studio/panels/user-scripts) to process messages on a specific topic and convert them to another type
- Create a [message converter extension](https://foxglove.dev/blog/announcing-message-converter-extensions-in-foxglove-studio) to process and convert all messages with the `geometry_msgs/PointStamped` type.

**Comment by vinnnyr** [comment]
For those who might be looking at this, I was able to create a UserScript to handle this.
```
import { Input, Message } from "./types";
import { buildRosMarker, MarkerTypes } from "./markers";

type GlobalVariables = { id: number };

export const inputs = ["/lookahead_point"];
export const output = "/studio_script/point_visualization";

type PointStamped = Message<"geometry_msgs/PointStamped">;
type Marker = Message<"visualization_msgs/Marker">;

export default function script(
  event: Input<"/lookahead_point">,
  globalVars: GlobalVariables
): Marker {
  const { header, point } = event.message as PointStamped;
  // Build a Marker message using the received PointStamped message
  const marker = buildRosMarker({
    type: MarkerTypes.SPHERE, // Use a sphere as the marker type,
    header: {
      frame_id: header.frame_id,
      stamp: header.stamp,
      seq: 0,
    },
    pose: {
      position: point,
      orientation: { x: 0, y: 0, z: 0, w: 1 }, // Set the orientation to default values
    },
    scale: { x: 1.0, y: 1.0, z: 1.0 }, // Adjust the scale as needed
    color: { r: 1, g: 0, b: 0, a: 0.9 }, // Set the color to red
  });

  return marker;
}
```

So we can close this out.

**Reply by jtbandes** [reply]
Thanks for sharing!
